src/cmd-buildextend-aws

```diff
@@ -15,16 +15,21 @@ from cosalib.cmdlib import run_verbose, write_json, sha256sum_file
 # Parse args and dispatch
 parser = argparse.ArgumentParser()
 parser.add_argument("--build", help="Build ID")
-parser.add_argument("--region", help="EC2 region",
-                    required=True)
-parser.add_argument("--bucket", help="S3 Bucket",
-                    required=True)
+parser.add_argument("--region", help="EC2 region")
+parser.add_argument("--bucket", help="S3 Bucket")
 parser.add_argument("--name-suffix", help="Suffix for name")
 parser.add_argument("--grant-user", help="Grant user launch permission",
                     nargs="*", default=[])
 parser.add_argument("--log-level", help="ore log level")
+parser.add_argument("--upload", action='store_true', default=False,
+                    help="Upload to AWS")
 args = parser.parse_args()
 
+if args.upload:
+    for arg in ['region', 'bucket']:
+        if not args.arg:
+            raise Exception("Must provide region and bucket args to upload")
+
```

Comment:
This is unfortunate; I'm a bit surprised `argparse` doesn't have a fetaure for this semantic already.

Reply:
yeah. it's because they are only required if another argument is required IIUC